p2p/src/main/java/bisq/network/p2p/network/TorNetworkNode.java

```diff
@@ -115,7 +117,17 @@ protected Socket createSocket(NodeAddress peerNodeAddress) throws IOException {
     public Socks5Proxy getSocksProxy() {
         try {
             tor = Tor.getDefault();
-            return tor != null ? tor.getProxy() : null;
+
+            String stream = "";
+            if (streamIsolation) {
+                // create a random string
+                byte[] bytes = new byte[512]; // note that getProxy does Sha256 that string anyways
+                new SecureRandom().nextBytes(bytes);
+                stream = new String(bytes);
```

Comment:
Could there be any issues to use plain bytes here? Maybe more safe to use hex?

Reply:
Netlayer does a `stream.getBytes()` immediately, so there should be no issue. Anyhow, the `steam` now gets a Base64-encoded `bytes`.